rate.go

```diff
@@ -72,39 +85,70 @@ func (l *Limiter) AllowRate(name string, rateLimit timerate.Limit) (delay time.D
 	if rateLimit == timerate.Inf {
 		return 0, true
 	}
-
+	
 	dur := time.Second
 	limit := int64(rateLimit)
 	if limit == 0 {
 		limit = 1
 		dur *= time.Duration(1 / rateLimit)
 	}
-
+	
 	now := time.Now()
 	slot := now.UnixNano() / dur.Nanoseconds()
 	allow = l.fallbackLimiter.Allow()
-
+	
 	name = fmt.Sprintf("%s:%s-%d-%d", redisPrefix, name, dur, slot)
 	count, err := l.incr(name, dur, 1)
 	if err == nil {
 		allow = count <= limit
 	}
-
+	
 	if !allow {
 		delay = time.Duration(slot+1)*dur - time.Duration(now.UnixNano())
 	}
-
+	
 	return delay, allow
 }
 
+func (l *Limiter) delete(names ...string) (err error) {
+	var del *redis.IntCmd
+	_, err = l.redis.Pipelined(func(pipe *redis.Pipeline) error {
+		del = pipe.Del(names...)
+		return nil
+	})
+	
+	if err != nil {
+		return
+	}
+	
+	err = del.Err()
+	
+	return
+}
+
+func (l *Limiter) keys(name string) ([]string, error) {
+	var keys *redis.StringSliceCmd
+	_, err := l.redis.Pipelined(func(pipe *redis.Pipeline) error {
+		//hard coded since we know the format
+		keys = pipe.Keys("rate:" + name + "*")
+		return nil
+	})
```

Comment:
My thought for building this was, what if in a unit test you want to test rate limiting on a test key, then using the same key test all error codes on an api. It will cause assertion errors as you exceeded the limit.